src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -46,25 +48,35 @@ public function append($id, DomainEventStreamInterface $eventStream)
     {
         $id = (string) $id;
 
-        if (! isset($this->events[$id])) {
-            $this->events[$id] = [];
+        $aggregateEvents = [];
+        if (isset($this->events[$id])) {
+            $aggregateEvents = $this->events[$id];
         }
 
         foreach ($eventStream as $event) {
-            $playhead = $event->getPlayhead();
-            $this->assertPlayhead($this->events[$id], $playhead);
-
-            $this->events[$id][$playhead] = $event;
+            $aggregateEvents = $this->insertEvent($aggregateEvents, $event, $eventStream);
```

Comment:
Makes sense that leaves 2 options:
- Assert the complete stream instead of the event.
- Keep it the way it was, however the stream would be an extra argument to assertPlayhead.

While I'm at it I can also move the "transactional"-ish behaviour in another commit if you like.